test/test-parse-statement.cpp

```diff
@@ -133,7 +133,7 @@ TEST(test_parse, return_statement_disallows_newline) {
   for (const char8* second_line : {
            u8"!true",
            u8"'string'",
-           u8"() => {}",
+           u8"(() => {})",
```

Comment:
I think this change might reduce test coverage. I think the code checks for both arrow functions and parenthesized expressions, and your change removes coverage for arrow functions.

IMO, the following code should report one warning (E0179), not two warnings (E0179 and E0211):
```js
return
  () => 42;
```

Therefore, I think we should undo this change and make the test pass as-is.